### PR TITLE
thread `new_package_waiting_period` through to AutoMerge comment

### PR DESCRIPTION
**Pull request by ericphanson**
Based on #548 which should get merged first

closes  https://github.com/JuliaRegistries/RegistryCI.jl/issues/413, which is a recurring source of confusion on our private registry that uses a much lower number here

### CODE REVIEW

**Comment by ericphanson** [comment]
I'm so confused about
```
ERROR: LoadError: LoadError: LoadError: syntax: invalid keyword argument syntax "new_package_waiting_period"
Stacktrace:
 [1] top-level scope at /home/runner/work/RegistryCI.jl/RegistryCI.jl/src/AutoMerge/pull_requests.jl:174
```

File: src/AutoMerge/util.jl [hunk 1]
```diff
@@ -153,7 +153,7 @@ function _comment_noblock(n)
 end
 
 function comment_text_pass(
-    ::NewVersion, suggest_onepointzero::Bool, version::VersionNumber, is_jll::Bool
+    ::NewVersion, suggest_onepointzero::Bool, version::VersionNumber, is_jll::Bool; new_package_waiting_period=nothing
```
Comment (DilumAluthge) [inline src/AutoMerge/util.jl:156]: Why is a default value needed here?

**Comment by DilumAluthge** [comment]
> I'm so confused about
> 
> ```
> ERROR: LoadError: LoadError: LoadError: syntax: invalid keyword argument syntax "new_package_waiting_period"
> Stacktrace:
>  [1] top-level scope at /home/runner/work/RegistryCI.jl/RegistryCI.jl/src/AutoMerge/pull_requests.jl:174
> ```

Do you have something of the form `f(; new_package_waiting_period)`?

IIRC, that doesn't work on Julia 1.3. You'll need to do `f(; new_package_waiting_period=new_package_waiting_period)` instead.

**Comment by GunnarFarneback** [comment]
Yes, the kwarg shortcut arrived in Julia 1.5.

File: src/AutoMerge/pull_requests.jl [hunk 2]
```diff
@@ -231,7 +232,8 @@ function pull_request_build(data::GitHubAutoMergeData; check_license)::Nothing
             data.registration_type,
             data.suggest_onepointzero,
             data.version,
-            this_pr_can_use_special_jll_exceptions,
+            this_pr_can_use_special_jll_exceptions;
+            new_package_waiting_period
```
Comment (ericphanson) [inline src/AutoMerge/pull_requests.jl:236]:
```suggestion
            new_package_waiting_period=new_package_waiting_period
```

**Comment by ericphanson** [comment]
Yeah, I knew that... but still left one in. It was very confusing because the stacktrace pointed to the wrong file I think. It said
```
 [1] top-level scope at /home/runner/work/RegistryCI.jl/RegistryCI.jl/src/AutoMerge/pull_requests.jl:174
```
but that line seemed fine.

File: src/AutoMerge/util.jl (hunk 1 above)
Comment (ericphanson) [reply src/AutoMerge/util.jl:156]: I don't think it is
```suggestion
    ::NewVersion, suggest_onepointzero::Bool, version::VersionNumber, is_jll::Bool; new_package_waiting_period
```

File: src/AutoMerge/util.jl [hunk 3]
```diff
@@ -171,7 +171,7 @@ function comment_text_pass(
 end
 
 function comment_text_pass(
-    ::NewPackage, suggest_onepointzero::Bool, version::VersionNumber, is_jll::Bool
+    ::NewPackage, suggest_onepointzero::Bool, version::VersionNumber, is_jll::Bool; new_package_waiting_period=nothing
```
Comment (ericphanson) [inline src/AutoMerge/util.jl:174]:
```suggestion
    ::NewPackage, suggest_onepointzero::Bool, version::VersionNumber, is_jll::Bool; new_package_waiting_period
```

**Pull request merged**